src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs

```diff
@@ -761,6 +761,86 @@ public class C
             await VerifyCSharpAsync(source, shippedText, unshippedText, expected);
         }
 
+        [Fact]
+        [WorkItem(4584, "https://github.com/dotnet/roslyn-analyzers/issues/4584")]
+        public async Task DuplicateSymbolUsingObliviousInSameApiFiles()
+        {
+            var source = @"
+public class C
+{
+    public int Field;
+    public int Property { get; set; }
+}
+";
+
+            var shippedText = @"#nullable enable
+C
+C.C() -> void
+C.Field -> int
+C.Property.get -> int
+C.Property.set -> void
+{|RS0025:~C.Property.get -> int|}
+";
+
+            var unshippedText = @"";
+
+            await VerifyCSharpAsync(source, shippedText, unshippedText);
+        }
+
+        [Fact]
+        [WorkItem(4584, "https://github.com/dotnet/roslyn-analyzers/issues/4584")]
+        public async Task DuplicateSymbolUsingObliviousInDifferentApiFiles()
+        {
+            var source = @"
+public class C
+{
+    public int Field;
+    public int Property { get; set; }
+}
+";
+
+            var shippedText = @"#nullable enable
+C
+C.C() -> void
+C.Field -> int
+~C.Property.get -> int
+C.Property.set -> void
+";
+
+            var unshippedText = @"#nullable enable
+{|RS0025:C.Property.get -> int|}";
+
+            await VerifyCSharpAsync(source, shippedText, unshippedText);
+        }
+
+        [Fact]
+        [WorkItem(4584, "https://github.com/dotnet/roslyn-analyzers/issues/4584")]
+        public async Task MultipleDuplicateSymbolsUsingObliviousInDifferentApiFiles()
+        {
+            var source = @"
+public class C
+{
+    public int Field;
+    public int Property { get; set; }
+}
+";
+
+            var shippedText = @"#nullable enable
+C
+C.C() -> void
+C.Field -> int
+C.Property.get -> int
+C.Property.set -> void
+";
+
+            var unshippedText = @"#nullable enable
+{|RS0025:~C.Property.get -> int|}
+{|RS0025:C.Property.get -> int|}
+{|RS0025:~C.Property.set -> void|}";
+
+            await VerifyCSharpAsync(source, shippedText, unshippedText);
+        }
```

Comment:
Consider including a test with duplicates that both have the oblivious marker.

Reply:
Added the additional test to cover this case